packages/fileeditor/src/searchprovider.ts

```diff
@@ -20,15 +21,34 @@ export type FileEditorPanel = MainAreaWidget<FileEditor>;
  * File editor search provider
  */
 export class FileEditorSearchProvider
-  extends CodeMirrorSearchProvider
+  extends EditorSearchProvider<CodeEditor.IModel>
   implements ISearchProvider
 {
   /**
    * Constructor
    * @param widget File editor panel
    */
-  constructor(widget: FileEditorPanel) {
-    super(widget.content.editor as CodeMirrorEditor);
+  constructor(protected widget: FileEditorPanel) {
+    super();
+  }
+
+  isReadOnly = false;
```

Comment:
Should we read that information from the editor (in `EditorSearchProvider`):

```suggestion
  get isReadOnly(): boolean {
    return this.editor.getOption('readOnly');
  }
```

Reply:
Note to self: maybe add a test for this one.